stumpwm.lisp

```diff
@@ -227,21 +245,24 @@ The action is to call FUNCTION with arguments ARGS."
   (declare (ignore io-loop))
   (ccl::stream-device channel :input))
 
-#+sbcl
 (defun call-in-main-thread (fn)
-  (with-lock-held ((request-channel-lock *request-channel*))
-    (push fn (request-channel-queue *request-channel*)))
-  (let ((out (request-channel-out *request-channel*)))
-    ;; For now, just write a single byte since all we want is for the
-    ;; main thread to process the queue. If we want to handle
-    ;; different types of events, we'll have to change this so that
-    ;; the message sent indicates the event type instead.
-    (write-byte 0 out)
-    (finish-output out)))
-
-#-sbcl
-(defun call-in-main-thread (fn)
-  (run-with-timer 0 nil fn))
+  (if *in-main-thread*
+      (funcall fn)
+      ;; ELSE: Schedule the job in the main thread
+      (progn
+        #+call-in-main-thread
+        (progn
```

Comment:
I don't think merging these two definitions, ending up with an `if` containing two nested `progn`s, is a good idea. Keeping the two separate definitions, both with the same (simple) condition, and/or using `cond`, seems like a better idea.